gateway/gateway/service.py

```diff
@@ -22,6 +22,59 @@ class GatewayService(object):
     orders_rpc = RpcProxy('orders')
     products_rpc = RpcProxy('products')
 
+    @http(
+        "GET", "/products/<string:product_id>",
+        expected_exceptions=ProductNotFound
+    )
+    def get_product(self, request, product_id):
+        """Gets product by `product_id`
+        """
+        product = self.products_rpc.get(product_id)
+        return Response(
+            ProductSchema().dumps(product).data,
+            mimetype='application/json'
+        )
+
+    @http(
+        "POST", "/products",
+        expected_exceptions=(ValidationError, BadRequest)
+    )
+    def create_product(self, request):
```

Comment:
interestingly, this also allows updates to a product (as we just set the data in redis) 😄